src/packages/core/components/icon/icon.element.ts

```diff
@@ -16,29 +16,48 @@ export class UmbIconElement extends UmbLitElement {
 	@state()
 	private _color?: string;
 
+	@state()
+	private _fallbackColor?: string;
+
+	/**
+	 * Color alias or a color code directly.\
+	 * If a color has been set via the name property, this property will override it.
+	 * */
 	@property({ type: String })
 	public set color(value: string) {
-		if (!value) return;
-		this.#setColorStyle(value);
+		if (value) {
+			this.#setColorStyle(value, true);
+		} else {
+			this._color = undefined;
+		}
 	}
 	public get color(): string {
 		return this._color ?? '';
 	}
 
-	#setColorStyle(value: string) {
+	#setColorStyle(value: string, dominant = false) {
 		const alias = value.replace('color-', '');
 		const variable = extractUmbColorVariable(alias);
-		this._color = alias ? (variable ? `--uui-icon-color: var(${variable})` : `--uui-icon-color: ${alias}`) : undefined;
+		const color = alias ? (variable ? `--uui-icon-color: var(${variable})` : `--uui-icon-color: ${alias}`) : undefined;
+		if (dominant) {
+			this._color = color;
+		} else {
+			this._fallbackColor = color;
+		}
```

Comment:
Consider returning the color string instead, and let the consumers set it into whichever state property they would like. I think the it would be much more obvious what happens and when, if you did that.